src/components/invoice-form.tsx

```diff
@@ -38,348 +53,821 @@ interface InvoiceFormProps {
+            <Alert variant="warning">
+              <Terminal className="h-4 w-4" />
+              <AlertTitle>
+                Warning: You are creating an invoice with real funds
+              </AlertTitle>
+              <AlertDescription className="space-y-2">
+                <p>
+                  You've selected{" "}
+                  <span className="font-bold">
+                    {formatCurrencyLabel(form.watch("invoiceCurrency"))}
+                  </span>
+                  , which operates on a mainnet blockchain and uses real
+                  cryptocurrency.
+                </p>
+                <p>
+                  EasyInvoice is a demonstration app only, designed to showcase
+                  Request Network API functionality. Do not use for real
+                  invoicing.
+                </p>
+              </AlertDescription>
+            </Alert>
+          </div>
+        )}
+
+        {/* Only show payment currency selector for USD invoices */}
+        {form.watch("invoiceCurrency") === "USD" && (
+          <div className="space-y-2">
+            <Label htmlFor="paymentCurrency">Payment Currency</Label>
+            <Select
+              onValueChange={(value) => form.setValue("paymentCurrency", value)}
+              defaultValue={form.getValues("paymentCurrency")}
+            >
+              <SelectTrigger>
+                <SelectValue placeholder="Select payment currency" />
+              </SelectTrigger>
+              <SelectContent>
+                {getPaymentCurrenciesForInvoice("USD").map((currency) => (
+                  <SelectItem key={currency} value={currency}>
+                    {formatCurrencyLabel(currency)}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+            {form.formState.errors.paymentCurrency && (
+              <p className="text-sm text-red-500">
+                {form.formState.errors.paymentCurrency.message}
+              </p>
+            )}
+          </div>
+        )}
+
+        <div className="space-y-2">
+          <Label htmlFor="walletAddress">Your Wallet Address</Label>
+          <Input
+            {...form.register("walletAddress")}
+            placeholder="Enter your wallet address"
+          />
+          {form.formState.errors.walletAddress && (
+            <p className="text-sm text-red-500">
+              {form.formState.errors.walletAddress.message}
+            </p>
+          )}
+        </div>
+
+        <div className="flex items-center space-x-2">
+          <Checkbox
+            id="isCryptoToFiatAvailable"
+            checked={form.watch("isCryptoToFiatAvailable")}
+            onCheckedChange={(checked) => {
+              form.setValue("isCryptoToFiatAvailable", checked === true);
+            }}
+          />
+          <Label htmlFor="isCryptoToFiatAvailable">
+            Allow payment via bank account (crypto-to-fiat conversion)
```

Comment:
Further clarify that it's the _payee's_ bank account. Avoid the word "conversion" as that's typically reserved for "price/rate conversion" in the RN terminology.

```suggestion
            Allow payment to your bank account (Crypto to Fiat payment)
```